### PR TITLE
Install better

### PR DESCRIPTION
**Pull request by bobturneruk**
Improves install process and instructions. Uses `conda` to maintain environment and install `jupyter`, but `pip` to handle package requirements.

### CODE REVIEW

**Comment by bobturneruk** [comment]
Can someone from the RSE team to have a glance at this, please?

File: setup.py [hunk 1]
```diff
@@ -10,5 +10,5 @@
   download_url = 'https://github.com/lionfish0/calibration.git',
   keywords = ['calibration','sensors','internet of things','iot','network','pollution','gaussian process','variational inference','tensorflow'],
   classifiers = [],
-  install_requires=['numpy'],
+  install_requires=['numpy==1.18.0','tensorflow','absl-py','tensorflow-probability','pandas','scipy','gpflow'],
```
Comment (drj11) [inline setup.py:13]: and now also setuptools (which is not bundled with Python)

Comment (willfurnass) [inline setup.py:13]: Pinning to such a specific version of numpy could be challenging if wanting to use this pkg alongside other things. Could a bounded range be used instead?

**Review by drj11 — APPROVED** [review]
I've haven't actually run the instructions, but this is a clear improvement, so it should be merged.

File: setup.py (hunk 1 above)
Comment (drj11) [reply setup.py:13]: good idea. `1.18`

Comment (bobturneruk) [reply setup.py:13]: The requirement is tensorflow - I'll find the most relaxed bounds.

Comment (bobturneruk) [reply setup.py]: I'm struggling a bit. It seems not to want to install tensorflow now (although I can pip install it). I'll look into it but wold appreciate any advice:

```
(base) PS C:\Users\Bob Turner\Documents\calibration> python setup.py develop
running develop
running egg_info
writing calibration.egg-info\PKG-INFO
writing dependency_links to calibration.egg-info\dependency_links.txt
writing requirements to calibration.egg-info\requires.txt
writing top-level names to calibration.egg-info\top_level.txt
reading manifest file 'calibration.egg-info\SOURCES.txt'
writing manifest file 'calibration.egg-info\SOURCES.txt'
running build_ext
Creating c:\programdata\anaconda3\lib\site-packages\calibration.egg-link (link to .)
Adding calibration 1.0 to easy-install.pth file

Installed c:\users\bob turner\documents\calibration
Processing dependencies for calibration==1.0
Searching for gpflow
Reading https://pypi.org/simple/gpflow/
Downloading https://files.pythonhosted.org/packages/8e/90/d4a20a3542e595e5a7b86437730c3bc03504a1153f49628aee79634bc87b/gpflow-2.1.3-py3-none-any.whl#sha256=2ae1c0657deefdbf1806c7e366e1aafae74dd109bf316584c89163657ce4d62c
Best match: gpflow 2.1.3
Processing gpflow-2.1.3-py3-none-any.whl
Installing gpflow-2.1.3-py3-none-any.whl to c:\programdata\anaconda3\lib\site-packages
writing requirements to c:\programdata\anaconda3\lib\site-packages\gpflow-2.1.3-py3.7.egg\EGG-INFO\requires.txt
Adding gpflow 2.1.3 to easy-install.pth file

Installed c:\programdata\anaconda3\lib\site-packages\gpflow-2.1.3-py3.7.egg
Searching for tensorflow-probability
Reading https://pypi.org/simple/tensorflow-probability/
Downloading https://files.pythonhosted.org/packages/b3/99/07d4220b021a5d6f6b3de8872f2a634389ad224f8508258452e57aaaf328/tensorflow_probability-0.11.1-py2.py3-none-any.whl#sha256=e9ce5288ccdb80b6e311d1a2904b5af618a7cd5791d79d6fd051a628087ec0ae
Best match: tensorflow-probability 0.11.1
Processing tensorflow_probability-0.11.1-py2.py3-none-any.whl
Installing tensorflow_probability-0.11.1-py2.py3-none-any.whl to c:\programdata\anaconda3\lib\site-packages
writing requirements to c:\programdata\anaconda3\lib\site-packages\tensorflow_probability-0.11.1-py3.7.egg\EGG-INFO\requires.txt
Adding tensorflow-probability 0.11.1 to easy-install.pth file

Installed c:\programdata\anaconda3\lib\site-packages\tensorflow_probability-0.11.1-py3.7.egg
Searching for absl-py
Reading https://pypi.org/simple/absl-py/
Downloading https://files.pythonhosted.org/packages/b9/07/f69dd3367368ad69f174bfe426a973651412ec11d48ec05c000f19fe0561/absl_py-0.10.0-py3-none-any.whl#sha256=ea07d7d437798bffc14f39fccec3909d251a1e76e233205ded72b71c267e0178
Best match: absl-py 0.10.0
Processing absl_py-0.10.0-py3-none-any.whl
Installing absl_py-0.10.0-py3-none-any.whl to c:\programdata\anaconda3\lib\site-packages
writing requirements to c:\programdata\anaconda3\lib\site-packages\absl_py-0.10.0-py3.7.egg\EGG-INFO\requires.txt
Adding absl-py 0.10.0 to easy-install.pth file

Installed c:\programdata\anaconda3\lib\site-packages\absl_py-0.10.0-py3.7.egg
Searching for tensorflow
Downloading https://files.pythonhosted.org/packages/01/51/9e7da2c6197c7efc9ebcab944d71058b95aeae163810dcd7ac31d9656e3e/tensorflow-probability-0.2.0.tar.gz#sha256=1ef13fb6c65c90ed283caa35facb60fa4271e527a446215093e0d7b0cd891cda
Best match: tensorflow probability-0.2.0
Processing tensorflow-probability-0.2.0.tar.gz
Writing C:\Users\BOBTUR~1\AppData\Local\Temp\easy_install-stoykohc\tensorflow-probability-0.2.0\setup.cfg
Running tensorflow-probability-0.2.0\setup.py -q bdist_egg --dist-dir C:\Users\BOBTUR~1\AppData\Local\Temp\easy_install-stoykohc\tensorflow-probability-0.2.0\egg-dist-tmp-ytf7evna
warning: no files found matching 'BUILD'
warning: no files found matching 'BUILD' under directory 'tensorflow_probability'
creating c:\programdata\anaconda3\lib\site-packages\tfp_nightly-0.2.0.dev20201022-py3.7.egg
Extracting tfp_nightly-0.2.0.dev20201022-py3.7.egg to c:\programdata\anaconda3\lib\site-packages
Adding tfp-nightly 0.2.0.dev20201022 to easy-install.pth file

Installed c:\programdata\anaconda3\lib\site-packages\tfp_nightly-0.2.0.dev20201022-py3.7.egg
error: The 'tensorflow' distribution was not found and is required by calibration
```

Comment (bobturneruk) [reply setup.py]: It's weirdly trying to install `tensorflow_probability` instead of `tensorflow`.

**Comment by bobturneruk** [comment]
@lionfish0 - this is now ready - I've tested that `demo.ipynb` runs in this environment. I'd suggest squashing the merge.

**Pull request merged**